budget-service/src/main/java/org/java/mentorship/budget/controller/BudgetController.java

```diff
@@ -61,4 +61,12 @@ public ResponseEntity<Budget> deleteBudget(@PathVariable("id") Integer id) {
         Budget budget = BudgetContractMapper.entityToContract(deletedBudget);
         return ResponseEntity.ok(budget);
     }
+
+    @GetMapping("/user/{userId}")
```

Comment:
Not quite
If you look at the mapping from a REST perspective, this endpoint would return a single user with the given ID, not a list of Budgets.

Instead, this should be part of the getAllBudgets.
The userId comes in as a query param and lets you "filter" on the budgets: get the budgets for *this* user
An example for this can be found in the class NotificationController